internal/database/database.go

```diff
@@ -0,0 +1,49 @@
+package database
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"os"
+
+	_ "modernc.org/sqlite" // Import the sqlite3 driver
+)
+
+type Service struct {
+	DB *sql.DB
+}
+
+func Conect(DBFile string) (*Service, error) {
+	var install bool
+
+	if _, err := os.Stat(DBFile); os.IsNotExist(err) {
+		fmt.Println(DBFile)
+		install = true
+	}
+
+	// Open the database
+	db, err := sql.Open("sqlite", DBFile)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the table if it doesn't exist
+	if install {
+		sqlFile, err := os.ReadFile("./migration/create_table.sql")
+		if err != nil {
+			return nil, fmt.Errorf("failed to read SQL file: %v", err)
```

Comment:
А дальше увидел оборачивание, но нужно использовать `%w` (wrpa), чтобы ошибка не превращалась в текст и с ней можно было работать.
```suggestion
			return nil, fmt.Errorf("failed to read SQL file: %w", err)
```